### PR TITLE
Remove blast plaintext support from Bio.Blast

### PR DESCRIPTION
**Pull request by bow**
Continuing from the mailing list discussion [here](https://lists.open-bio.org/pipermail/biopython/2018-November/016572.html), this PR would remove plain text parsing functionalities (which have been officially deprecated) into Bio.SearchIO._legacy, to be used by `Bio.SearchIO`'s `blast-text` format.

This is still a WIP, as the documentation probably still needs to be updated. The proposed essential change is the first commit of this PR though, so you can already see if this is sensible or not.

- [x] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [x] I have read the ``CONTRIBUTING.rst`` file and understand that AppVeyor and
TravisCI will be used to confirm the Biopython unit tests and ``flake8`` style
checks pass with these changes.

- [x] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Bio/SearchIO/_legacy/__init__.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
```
Comment (peterjc) [inline Bio/SearchIO/_legacy/__init__.py]: Stray blank line at the start

**Comment by peterjc** [comment]
This would need an update to ``DEPRECATED.rst`` so say ``Bio.Blast.NCBIStandalone`` and `` Bio.ParserSupport`` have been removed.

Also the tests would need their imports updated to use the new (private) location under the SearchIO legacy folder.

But it looks good in principle 👍

**Comment by mdehoon** [comment]
Looks good to me, once we can fix the AppVeyor and Travis errors.

**Comment by bow** [comment]
All right, I've updated the docs, tests, and `DEPRECATED.rst`. Let's see if all the tests pass, and then I will merge this no sooner than 9 Dec 17.00 UTC if there are no objections.

**Comment by peterjc** [comment]
I think you need to add ``Bio.SearchIO._legacy`` to ``setup.py``, and if that works, I would suggest doing a squash-and-merge?

**Comment by bow** [comment]
Ok ~ I just pushed the change. I'm also removing the WIP prefix. Let's see if this works.

**Comment by peterjc** [comment]
Green light :)

Do you want to keep the commit history, or just squash-and-merge this?

**Comment by bow** [comment]
Squash-and-merge seems more sensible ~ should be done anytime now :).

**Comment by peterjc** [comment]
Thank you!

**Pull request merged**